src/apps/views/AppView/AppView.tsx

```diff
@@ -39,12 +42,16 @@ export const AppView: React.FC<AppProps> = ({ id }) => {
     return <NotFoundPage onBack={() => navigate(AppPaths.appListPath)} />;
   }
 
-  const appCompleteUrl = AppUrls.resolveAppCompleteUrlFromDashboardUrl(
+  let appCompleteUrl = AppUrls.resolveAppCompleteUrlFromDashboardUrl(
     location.pathname,
     data?.app?.appUrl || "",
     id,
   );
 
+  if(appPath) {
+    appCompleteUrl = `${appCompleteUrl}/${appPath}`;
```

Comment:
You are right but it will require a refactor (this method is already broken because it accepts optional params).

I will refactor entire URLs segment separately